### PR TITLE
Introduce the client declaration

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose

```ballerina
const map<anydata> ClientAnnot on source client;

client "http://www.example.com/apis/one.yaml" as foo;

public function main() {
    foo:client x;

    @ClientAnnot
    client "http://www.example.com/apis/two.yaml" as bar;
    bar:client y;
}
```

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/36835
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/37199
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/37359 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java [hunk 1]
```diff
@@ -179,6 +179,9 @@ public List<String> skipList() {
                 "func_type_source_15.bal", "func_type_source_16.bal", "import_decl_source_24.bal",
                 "member_access_expr_source_11.bal", "float_literal_source_08.bal", "object_type_def_source_47.bal", 
                 "client_resource_access_action_source_05.bal", "client_resource_access_action_source_06.bal", 
+                "resiliency_source_05.bal", "client_decl_stmt_source_08.bal", "client_decl_stmt_source_13.bal",
```
Comment (MaryamZi) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java:182]:
```suggestion
                "client_decl_stmt_source_08.bal", "client_decl_stmt_source_13.bal",
```

**Review by malinthar — APPROVED** [review]

**Review by dulajdilshan — APPROVED** [review]

**Review by azinneera — APPROVED** [review]

**Review by gayalkuruppu — APPROVED** [review]

**Review by hasithaa — APPROVED** [review]

File: .github/workflows/pull_request_ubuntu_build.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ on:
       - stage-swan-lake
       - 2201.[0-9]+.x
       - 2201.[0-9]+.[0-9]+-stage
+      - client-declaration
```
Comment (MaryamZi) [inline .github/workflows/pull_request_ubuntu_build.yml:14]: Will send a separate PR removing these.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/statements/BLangClientDeclarationStatement.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ *
+ * @since 2201.3.0
+ */
+public class BLangClientDeclarationStatement extends BLangStatement implements ClientDeclarationStatementNode {
+
+    public BLangClientDeclaration clientDeclaration;
+
+    @Override
+    public void setClientDeclaration(ClientDeclarationNode xmlnsDecl) {
+        this.clientDeclaration = (BLangClientDeclaration) xmlnsDecl;
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/statements/BLangClientDeclarationStatement.java:39]: Will fix param name in a separate PR.
